db/user_test.go

```diff
@@ -14,6 +15,83 @@ type SyncNewUsersTestCase struct {
     sendEmails bool
 }
 
+type ResolveUsersTestCase struct {
+    input []string
+    expectedOutput []string
+    course *model.Course
+    actualOutput []string
+    err error
+}
+
+func (this *DBTests) DBTestResolveUsers(test *testing.T) {
+    defer ResetForTesting();
+
+    for i, testCase := range getResolveUsersTestCases() {
+        ResetForTesting();
+        if (i == 4) {
+            course := MustGetCourse(TEST_COURSE_ID);
+            SaveUser(course, model.NewUser("second_student@test.com", "Billy Bob", model.RoleStudent));
+            RemoveUser(course, "owner@test.com");
+        }
+        testCase.actualOutput, testCase.err = ResolveUsers(testCase.course, testCase.input);
+        if (testCase.err != nil) {
+            test.Errorf("Case %d (%+v): Resolve User failed: '%v'.", i, testCase, testCase.err);
+            continue;
+        }
+        if (!reflect.DeepEqual(testCase.expectedOutput, testCase.actualOutput)) {
+            test.Errorf("Case %d (%+v): Incorrect Output. Expected: '%v', actual '%v'.", i,
+                testCase, testCase.expectedOutput, testCase.actualOutput);
+            continue;
+        }
+    }
+}
+
+
+func getResolveUsersTestCases() ([]ResolveUsersTestCase) {
+    course := MustGetCourse(TEST_COURSE_ID);
+    testCases := []ResolveUsersTestCase{
+        ResolveUsersTestCase{
+            input: make([]string, 0, 1),
+            course: nil,
+        },
+        ResolveUsersTestCase{
+            input: make([]string, 0, 3),
+            course: nil,
+        },
+        ResolveUsersTestCase{
+            input: make([]string, 0, 1),
+            course: course,
+        },
+        ResolveUsersTestCase{
+            input: make([]string, 0, 1),
+            course: course,
+        },
+        ResolveUsersTestCase{
+            input: make([]string, 0, 5),
+            course: course,
+        },
+        ResolveUsersTestCase{
+            input: make([]string, 0, 2),
+            course: course,
+        },
+    }
+    testCases[0].input = append(testCases[0].input, "");
```

Comment:
Just define the slices when you define the test case.
Doing it here like this makes it much harder to maintain (what if you want to insert a new test case after the first one?).